src/cloudai/cli/handlers.py

```diff
@@ -209,58 +184,85 @@ def handle_generate_report(args: argparse.Namespace) -> int:
     return 0
 
 
-def handle_verify_tests(args: argparse.Namespace) -> int:
-    """Verify the test configurations."""
-    root: Path = args.test_configs
+def expand_file_list(root: Path) -> tuple[int, List[Path]]:
     if not root.exists():
-        logging.error(f"Tests directory {root} does not exist.")
-        return 1
+        logging.error(f"{root} does not exist.")
+        return (1, [])
 
     test_tomls = [root]
     if root.is_dir():
         test_tomls = list(root.glob("*.toml"))
         if not test_tomls:
-            logging.error(f"No test tomls found in {root}")
-            return 1
+            logging.error(f"No TOMLs found in {root}")
+            return (1, [])
 
-    rc = 0
+    return (0, test_tomls)
+
+
+def handle_verify_systems(args: argparse.Namespace) -> int:
+    root: Path = args.system_configs
+    err, system_tomls = expand_file_list(root)
+    if err:
+        return err
+
+    nfailed = 0
+    for test_toml in system_tomls:
+        logging.info(f"Verifying {test_toml}...")
+        try:
+            Parser.parse_system(test_toml)
+        except Exception:
+            nfailed += 1
+
+    if nfailed:
+        logging.error(f"{nfailed} out of {len(system_tomls)} system configurations have issues.")
```

Comment:
Would it be better to print out the failed system TOML files instead of reporting only the numbers?